lib/src/rules/unnecessary_getters_setters.dart

```diff
@@ -89,21 +90,20 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
 
     // Only select getters with setter pairs
-    getters.keys
-        .where((id) => setters.keys.contains(id))
-        .forEach((id) => _visitGetterSetter(getters[id], setters[id]));
+    for (var id in getters.keys) {
+      _visitGetterSetter(getters[id]!, setters[id]);
+    }
   }
 
-  void _visitGetterSetter(
-      MethodDeclaration? getter, MethodDeclaration? setter) {
-    if (getter != null &&
-        setter != null &&
-        isSimpleSetter(setter) &&
+  void _visitGetterSetter(MethodDeclaration getter, MethodDeclaration? setter) {
+    if (setter == null) return;
```

Comment:
Unfortunately, `if (setter?.declaredElement == null) return;`  does not promote `setter` to non-null. There  is a language issue for this. So this reads nicer as is, rather than null-checking `setter` below.